Kernel/Process.cpp

```diff
@@ -3263,9 +3263,15 @@ int Process::sys$getsockname(int sockfd, sockaddr* addr, socklen_t* addrlen)
 
     auto& socket = *description->socket();
     REQUIRE_PROMISE_FOR_SOCKET_DOMAIN(socket.domain());
-    if (!socket.get_local_address(addr, addrlen))
+
+    // NOTE: Create a stack buffer the same size as the largest sockaddr type.
+    char addr_buffer[sizeof(sockaddr_un)];
+
+    if (!socket.get_local_address(reinterpret_cast<sockaddr*>(addr_buffer), &addrlen))
```

Comment:
Let's be a little more defensive about this and not pass `addrlen` into `get_local_address()` without clamping it to `sizeof(sockaddr_un)`.

Something like this before the call:
```cpp
addrlen = min(addrlen, sizeof(sockaddr_un);
```
Same comment for the other commit.

Reply:
Should we really clamp? Shouldn't we reject adddrlen's  larger than sockaddr_un?

Reply:
The `addrlen` argument tells us how many bytes are available at `addr`. Since we're changing the `addr` argument we pass into `Socket`, we should also pass an appropriate `addrlen`.

That said, now that we're making a copy *anyway*, it might make more sense to change the API of `Socket::get_local_address()` to have it return a stack-allocated value object that we could then copy out of `sys$getsockname()`. Then we wouldn't have to care about creating an appropriately sized buffer and hoping that nothing bad happens to it.